core/kernel/dt.c

```diff
@@ -51,6 +51,101 @@ bool dt_have_prop(const void *fdt, int offs, const char *propname)
 	return prop;
 }
 
+int dt_get_irq(void *fdt, int node)
+{
+	const uint32_t *int_prop;
+	int len_prop;
+	int it_num;
+
+	/*
+	 * Interrupt property is defined with 3x32 bits word:
+	 *  - Type of interrupt
+	 *  - Interrupt Number
+	 *  - Interrupt Trigger type
+	 */
+	int_prop = fdt_getprop(fdt, node, "interrupts", &len_prop);
+
+	if ((int_prop == NULL) || (len_prop < 2))
+		return -1;
+
+	it_num = fdt32_to_cpu(int_prop[1]);
+
+	return it_num;
+}
+
+paddr_t dt_node_offset_by_compatible_status(void *fdt, int status, int *node,
+		const char *compatible)
+{
+	paddr_t offset = 0;
+	int fnode;
+	int node_status;
+
+	fnode = fdt_node_offset_by_compatible(fdt, *node,
+			compatible);
+
+	while (fnode != -FDT_ERR_NOTFOUND) {
+		node_status = _fdt_get_status(fdt, fnode);
+		if ((node_status & status) == status) {
+			offset = _fdt_reg_base_address(fdt, fnode);
+			*node  = fnode;
+			break;
+		}
+
+		fnode = fdt_node_offset_by_compatible(fdt, fnode,
+				compatible);
+	}
+
+	return offset;
+}
+
+int dt_disable_status(void *fdt, int node)
+{
+	const char *prop;
+	int len;
+
+	prop = fdt_getprop(fdt, node, "status", &len);
+	if (!prop) {
+		/* status is not available, just add it */
+		if (fdt_setprop_string(fdt, node, "status", "disabled"))
+			return -1;
+	} else {
+		/* status is there, modify it */
+		/* Actually we do not have to put disabled, anything that is
+		 * not "ok" or "okay" will work
+		 */
+		if (fdt_setprop_inplace(fdt, node, "status",
+				"disa", len))
```

Comment:
I suppose you're using a truncated status value as an optimisation to avoid growing the FDT, you can mention that in the comment.
What if `len` is larger than `5`?

Reply:
I'll update the comment with more explanation. 
The goal is to modify the DT with a value not equal to "ok" or "okay". I'll modify the new setting value to be "disabled" because the fdt_setprop_inplace is going to truncate the value function of the "len" value and else copy the "disabled" at the beginning of the property and keep the rest of the original property.
Objectif of using this function is to not change the DT offset and so recalculate the overall DT. For me, this is a time consuming operation that is not necessary in this case.